### ISSUE TITLE
cache.disk sometimes stores mysql password in filesystem path

### ISSUE THREAD

**Issue by zenite** [web2py/web2py#820]
I noticed that one query creates a file system path (in linux) that contains the mysql password, such as myapp/cache/mysql:/user:pass@localhost/app/SELECT...

I'm quoting Leonel Câmara about the issue:

"This is definitely a bug in the DAL. Probably a very old one that was made evident by the new lockless cache implementation. Passwords should never have been saved in the cache in raw format as the cache is not supposed to be safe for that.  
  
BaseAdapter does this in the select

            (cache_model, time_expire) = cache
            key = self.uri + '/' + sql
            if len(key)>200: key = hashlib_md5(key).hexdigest()
            args = (sql,fields,attributes)

Basically it should always, always, use the hash, not just if the key's len is >200. Please submit an issue this must be fixed ASAP."

**Comment by gi0baro** [comment]
@niphlod let's keep this open until the next release of pydal/web2py with the fix. Ok with that?

**Comment by niphlod** [comment]
grrrrr: another thing to agree on how to manage. 
The way I see it, the bug is fixed in pydal. Next release of web2py will include a newer pydal incorporating the fix (i.e. at least 15.03, soon to be released). Since the fix is on pydal, and pydal specific, users living on the edge (such as the ones running web2py "master") can try pydal "master".

Anyway.... being annoyngly specific, it's still true that until web2py "master" incorporates the fix updating the reference to the submodule, it'll still be an open issue............
But, I fear that if we start to track intra-issues keeping them open until web2py incorporates a new "golden release", we'll surely break up/leave behind something

IMHO the issue is closed as soon as it's merged into pydal master.

**Comment by gi0baro** [comment]
I see your point.
I think that:
* users living on the edge can track the pydal master entering the folder inside web2py and issuing `git pull` and `git checkout master`
* any other user won't see this fixed until a new release, so *being annoyingly specific* as you said, for the *majority* of the user this is not actually fixed

Said that, I think we can:
* closing these issues considering that in the next release of web2py the fixes will be included
* decide if under **these specific circumstances** where the issue is *highly dangerous* we should update the pydal submodule to track the last commit with the fix.

What do you think?

**Comment by niphlod** [comment]
ehm.... *annoyingly specific* speaking :-P for other issues (i.e. only in web2py's code) people needs to fetch master to enjoy the fix, and we're still happy to close issues on web2py as soon as the fix gets into master.

quoting you, I'd go for:
- closing these issues considering that in the next release of web2py the fixes will be included
- decide if under **these specific circumstances** where the issue is *highly dangerous* we should *tag a newer golden release on pydal (e.g. 15.02.1)*, then update the pydal submodule to track the *golden release* with the fix.

**Comment by gi0baro** [comment]
+1 for the *nightly release* of pydal.
But since we're using the date format for release I'm proposing to release a 15.02.26 version today. I can do that in the next couple of hours.
